src/main/java/com/kobylynskyi/graphql/codegen/scala/ScalaGraphQLTypeMapper.java

```diff
@@ -72,11 +76,25 @@ public String wrapApiReturnTypeIfRequired(MappingContext mappingContext,
         if (computedTypeName.startsWith(SCALA_UTIL_LIST) &&
                 Utils.isNotBlank(mappingContext.getApiReturnListType())) {
             // in case it is query/mutation, return type is list and apiReturnListType is set
-            return computedTypeName.replace(SCALA_UTIL_LIST, mappingContext.getApiReturnListType());
+            if (mappingContext.getApiReturnListType().contains(MappingConfigConstants.API_RETURN_NAME_PLACEHOLDER)) {
+                Matcher matcher = SCALA_UTIL_LIST_ELEMENT_REGEX.matcher(computedTypeName);
+                matcher.find();
```

Comment:
@matsudamper I think it's better if you do `if (matcher.find()) { ... }`

Reply:
Implemented.
https://github.com/kobylynskyi/graphql-java-codegen/pull/1200/commits/2542b8f5de5bc5660372d7424c60c83b44e2b0c3